node/node.go

```diff
@@ -91,31 +99,67 @@ func NewNodeController(p NodeProvider, node *corev1.Node, nodes v1.NodeInterface
 	}
 
 	n.nodePingController = newNodePingController(n.p, n.pingInterval, n.pingTimeout)
+	if n.leaseControllerProvider != nil {
+		n.leaseController = n.leaseControllerProvider(n.pingInterval)
```

Comment:
Why is the controller overridden if it was previously set with `WithNodeEnableLeaseV1(client, 10)`, for instance?

Reply:
The controller isn't set, the With... options set the controller provider, which is then resolved here, because the lease controller must be instantiated after the ping controller

Reply:
Guess I need to spend more time w/ the code then, sorry.

Reply:
Yeah, let me add comments.

Reply:
there were already comments here: https://github.com/virtual-kubelet/virtual-kubelet/pull/934/files#diff-375d57e386f20eaa5f09f02bb9d28bfc48ac3dca18d0325f59492208219e5618R229